pandas/tests/resample/test_base.py

```diff
@@ -106,7 +106,9 @@ def test_resample_empty_series(freq, empty_series, resample_method):
     if isinstance(s.index, PeriodIndex):
         expected.index = s.index.asfreq(freq=freq)
     else:
-        expected.index = s.index._shallow_copy(freq=freq)
+        idx = s.index
```

Comment:
same comment as above, why not just collapse this so: ```s.index.asfreq(freq=freq)``` or ``s.copy()``

we don't need to have the shallow copy used like this in tests, nor work-arounds like this.

Reply:
implemented a helper _asfreq_compat for this, cleans up some code nicely